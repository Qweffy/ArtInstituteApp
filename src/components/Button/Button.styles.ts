import { StyleSheet, ViewStyle } from 'react-native'
import { PrimaryColors } from '../../styles/Colors'
import { bodyRegular } from '../../styles/Fonts'
import { mediumSize } from '../../styles/Size'

type ButtonStyleType = {
  buttonContainer: ViewStyle
  button: ViewStyle
  disabledButton: ViewStyle
  disabledLabel: ViewStyle
  buttonLabel: ViewStyle
  spinner: ViewStyle
}

export default StyleSheet.create<ButtonStyleType>({
  buttonContainer: {
    flex: 0,
    justifyContent: 'flex-start',
    flexDirection: 'row',
  },
  button: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: mediumSize,
    borderRadius: 4,
    minHeight: 56,
  },
  disabledButton: {
    backgroundColor: PrimaryColors.LightGray,
  },
  disabledLabel: {
    color: PrimaryColors.Black,
  },
  buttonLabel: {
    ...bodyRegular,
    color: PrimaryColors.White,
  },
  spinner: {
    color: PrimaryColors.White,
    marginRight: 8,
  },
})
