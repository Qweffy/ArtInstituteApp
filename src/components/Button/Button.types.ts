import { ViewStyle } from 'react-native'

export type ButtonInputProps = {
  isLoading?: boolean
}

export type ButtonFunctionProps = {}

export type ButtonOwnProps = {
  mainColor?: string
  disabled?: boolean
  label: string
  containerStyles?: ViewStyle
  buttonStyles?: ViewStyle
  labelStyles?: ViewStyle
  onPress: (() => {}) | (() => void)
}

export type ButtonProps = ButtonInputProps & ButtonFunctionProps & ButtonOwnProps
