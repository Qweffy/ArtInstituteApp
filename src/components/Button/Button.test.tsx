import { render, screen } from '@testing-library/react-native'
import React from 'react'

import { Button } from './Button'

describe('<Button/>', () => {
  test('<Button> renders correctly', () => {
    render(<Button label={'hola'} onPress={() => console.log('hola')} />)
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
