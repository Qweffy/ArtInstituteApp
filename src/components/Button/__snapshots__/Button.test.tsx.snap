// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> <Button> renders correctly 1`] = `
<View
  style={
    [
      {
        "flex": 0,
        "flexDirection": "row",
        "justifyContent": "flex-start",
      },
      {},
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "purple",
        "borderRadius": 4,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "minHeight": 56,
        "opacity": 1,
        "paddingVertical": 16,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "white",
            "fontFamily": "Open Sans",
            "fontSize": 16,
            "fontWeight": "600",
            "letterSpacing": 0,
            "lineHeight": 24,
          },
          {},
          {},
        ]
      }
    >
      hola
    </Text>
  </View>
</View>
`;
